packages/functions/src/__tests__/__helpers__/tester.ts

```diff
@@ -30,11 +30,26 @@ export default async function <O = unknown>(
     },
   });
 
-  const results = await s.run(input instanceof Document ? input : JSON.stringify(input));
-  return results
-    .filter(result => result.code === 'my-rule')
-    .map(error => ({
-      path: error.path,
-      message: error.message,
-    }));
+  try {
+    const results = await s.run(input instanceof Document ? input : JSON.stringify(input));
+    return results
+      .filter(result => result.code === 'my-rule')
+      .map(error => ({
+        path: error.path,
+        message: error.message,
+      }));
+  } catch (e: unknown) {
+    if (
+      e instanceof Error &&
+      Array.isArray((e as Error & { errors?: unknown }).errors) &&
+      (e as Error & { errors: unknown[] }).errors.length === 1
+    ) {
+      const actualError = (e as Error & { errors: [unknown] }).errors[0];
+      throw actualError instanceof Error && 'cause' in (actualError as Error & { cause?: unknown })
+        ? (actualError as Error & { cause: unknown }).cause
+        : actualError;
+    } else {
+      throw e;
+    }
+  }
```

Comment:
hopefully this can be simplified once `AggregateError` and error causes are somehow exposed in standard lib

Reply:
So what's going on here? Don't quite understand this change.

Reply:
it's to simplify testing.
Assertion of `AggregateError` is shallow, meaning nested `errors` are not checked.
This change makes it so we re-throw a single error so `toThrow` works as expected.
In 99% (if not 100%) of cases we have a single error thrown, so that's why we unpack that AggregateError.
Makes sense?

Reply:
Previously we threw a single exception when `AggregateErrors.errors.length === 1`, but I simplified it in the code.
This, however, meant we needed to move that logic in here, but it's a test helper.